Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,25 +336,135 @@ def onApplyButton(self):
+            contents = pk.load(f)
+
+        print(f"File type for {filename} is not currently supported")
+        print(f"{filename} contains {summary_repr(contents)}") # !!!
+        # !!! Create node from contents
+
+        return None
+
+    def createNode(self, filename, basename_repr, props):
+        # Determine the node type from the filename extension, using its
+        # immediate-ancestor directory's name if necessary.  Note: check for
+        # ".seg.nrrd" before checking for ".nrrd".
+        if filename.endswith(".seg.nrrd"):
+            node = slicer.util.loadSegmentation(filename, properties=props)
+            node.CreateClosedSurfaceRepresentation()
+        elif filename.endswith(".nrrd"):
+            directory = os.path.basename(os.path.dirname(filename))
+            if directory == "images":
+                node = slicer.util.loadVolume(filename, properties=props)
+                self.show_nodes.append(node)
+            elif directory == "segmentations_computed":
+                node = slicer.util.loadSegmentation(filename, properties=props)
+                node.CreateClosedSurfaceRepresentation()
+            else:
+                # Guess
+                node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".fcsv"):
+            node = slicer.util.loadMarkups(filename)
+        elif filename.endswith(".mha"):
+            node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".png"):
+            node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".p3"):
+            node = self.processP3(filename, properties=props)
+        elif filename.endswith(".xls"):
+            print(f"File type for {basename_repr} is not currently supported")
+            node = None
+        else:
+            print(f"File type for {basename_repr} is not recognized")
+            node = None
+        return node
+
+    def addOutputWidget(self, shNode, subject_item, filename):
+        # The node types supported by 3D Slicer generally can be found with fgrep
+        # 'loadNodeFromFile(filename' from
+        # https://github.com/Slicer/Slicer/blob/main/Base/Python/slicer/util.py.
+        # Currently they are AnnotationFile, ColorTableFile, FiberBundleFile,
+        # MarkupsFile, ModelFile, ScalarOverlayFile, SegmentationFile,
+        # ShaderPropertyFile, TableFile, TextFile, TransformFile, and VolumeFile.
+
+        basename = os.path.basename(filename)
+        basename_repr = repr(basename)
+        props = {"name": basename, "singleFile": True, "show": False}
+
+        node = self.createNode(filename, basename_repr, props)
+        if node is None:
+            return
+        node_item = shNode.GetItemByDataNode(node)
+        # The node item is assigned the subject item as its parent.
+        shNode.SetItemParent(node_item, subject_item)
+        return
+
+    def addOutputWidgets(self, list_of_files):
```

Comment:
This function also needs a more accurate name and probably could use a docstring.

Reply:
Done.